### ISSUE TITLE
Mention in docs. that Cloudflare workers cannot have anything initialized outside a function

### ISSUE THREAD

**Issue by prabal-rje** [sst/ion#1266]
Please mention somewhere in SST docs that initializing outside of *a* function in Cloudflare workers is going to cause issues.

```javascript
/*
This is merely an EXAMPLE to demo a problem - do NOT publish your secrets on a public API
*/

// CORRECT WAY

function get_some_secret() {
    return Resource.SomeSecret.value;
}

export default {
    async fetch(request: Request, env: any, ctx: any) {
        return new Response(JSON.stringify({"config_val": get_some_secret()})), {
            headers: { 'Content-Type': 'application/json'  }, 
        });  
    }, 
}; 

// WRONG WAY

const x = Resource.SomeSecret.value;

export default {
    async fetch(request: Request, env: any, ctx: any) {
        return new Response(JSON.stringify({"config_val": x})), {
            headers: { 'Content-Type': 'application/json'  }, 
        });  
    }, 
}; 
```

**Comment by prabal-rje** [comment]
Also the following issue - that the Cloudflare Types SDK (a dev dependency) must be installed for SST Link Bindings to work properly.

https://github.com/sst/ion/issues/958#issuecomment-2417475504

**Comment by prabal-rje** [comment]
DISCREPANCY 1:

This issue causes current Cloudflare examples in SST docs to be non-replicable - which is a bit of a concern.

https://discord.com/channels/983865673656705025/1084270849206526163/1291968701905502269

TL;DR: Linking D1 with Cloudflare Worker doesn't work anymore with the `link` paramter (as it does in SST docs and examples) and requires a weird workaround.

Cheers,
PG

**Comment by prabal-rje** [comment]
BUG 1:

Linking with `sst.cloudflare.Queue` simply doesn't work and returns the following error(s).

```JavaScript
export const queue = new sst.cloudflare.Queue("MyQueue"); 

export const worker = new sst.cloudflare.Worker("MyWorker", {
  handler: "packages/functions/src/some_worker_handler.ts",
  url: true,      
  link: [queue],          
});      
```

```Shell
~  Deploy

|  Error       MyWorkerScript cloudflare:index:WorkerScript
|  cloudflare:index/workerScript:WorkerScript resource 'MyWorker
Script' has a problem: Missing required argument. The argument "
queue_binding.0.binding" is required, but no definition was foun
d.. Examine values at 'MyWorkerScript.queueBindings'.
|  Error       MyWorkerScript cloudflare:index:WorkerScript
|  cloudflare:index/workerScript:WorkerScript resource 'MyWorker
Script' has a problem: Invalid or unknown key. Examine values at
 'MyWorkerScript.queueBindings'.

✕  Failed
   MyWorkerScript cloudflare:index:WorkerScript
   cloudflare:index/workerScript:WorkerScript resource 'MyWorker
```
